frontend/src/metabase/visualizations/echarts/cartesian/option/axis.ts

```diff
@@ -0,0 +1,158 @@
+import moment from "moment-timezone";
+import type { AxisBaseOptionCommon } from "echarts/types/src/coord/axisCommonTypes";
+import type { CartesianAxisOption } from "echarts/types/src/coord/cartesian/AxisModel";
+import type {
+  ComputedVisualizationSettings,
+  RemappingHydratedDatasetColumn,
+  RenderingContext,
+} from "metabase/visualizations/types";
+import type { CardSeriesModel } from "metabase/visualizations/echarts/cartesian/model/types";
+
+const getAxisNameDefaultOption = (
+  { getColor, fontFamily }: RenderingContext,
+  nameGap: number,
+  name?: string,
+): Partial<AxisBaseOptionCommon> => ({
+  name,
+  nameGap,
+  nameLocation: "middle",
+  nameTextStyle: {
+    color: getColor("text-dark"),
+    fontSize: 14,
+    fontWeight: 900,
+    fontFamily,
+  },
+});
+
+const getTicksDefaultOption = ({ getColor, fontFamily }: RenderingContext) => {
+  return {
+    hideOverlap: true,
+    color: getColor("text-dark"),
+    fontSize: 12,
+    fontWeight: 900,
+    fontFamily,
+  };
+};
+
+export const getXAxisType = (settings: ComputedVisualizationSettings) => {
+  switch (settings["graph.x_axis.scale"]) {
+    case "timeseries":
+      return "time";
+    case "linear":
+      return "value";
+    default:
+      // TODO: implement histogram
+      return "category";
+  }
+};
+
+export const buildDimensionAxis = (
+  settings: ComputedVisualizationSettings,
+  renderingContext: RenderingContext,
```

Comment:
Nit: let's make `renderingContext` the last argument for consistency with `buildAxes`. In general it would be nice for it to always be the last arg.